simulator/mips/mips_instr.cpp

```diff
@@ -216,12 +226,27 @@ FuncInstr::FuncInstr( uint32 bytes, Addr PC,
 void FuncInstr::initFormat()
 {
     bool is_R = ( instr.asR.opcode == 0x0);
+    bool is_RI = ( instr.asR.opcode == 0x1);
     uint8 ident = is_R ? instr.asR.funct : instr.asR.opcode;
 
-    for ( const auto& entry : isaTable)
+    if ( is_R)
+    {    
+        const auto& entry = isaMapR.at( ident);
+        if ( entry != isaMapR.end())
+        {
+            format    = entry.format;
+            operation = entry.operation;
+            mem_size  = entry.mem_size;
+            name      = entry.name.c_str();
```

Comment:
Why do you need `c_str()' here? The same code few lines below works perfectly without it.

Reply:
Sorry, I do not know where it's from

Reply:
Who knows?